### PR TITLE
Scale down and restore PX shared applications during upgrade

### LINKED ISSUE #13: Add support to drain and undrain all application pods using Portworx shared volumes
FEATURE REQUEST:  Drain and undrain all applications pods using Portworx shared volumes from all nodes. This can be used during major release upgrades of Portworx.

### PR DESCRIPTION
**Pull request by harsh-px**
Closes #13

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** (optional)
Closes #

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/cluster/px/px.go [hunk 1]
```diff
@@ -243,3 +256,217 @@ func (ops *pxClusterOps) runDockerPuller(imageToPull string) error {
...
+			if err != nil {
+				if matched, _ := regexp.MatchString(".+ not found", err.Error()); matched {
+					return nil, false, nil // done as deployment is deleted
+				}
+
+				return nil, true, err
+			}
+
+			dCopy = dCopy.DeepCopy()
+			dCopy.Annotations[replicaMemoryKey] = fmt.Sprintf("%d", *dCopy.Spec.Replicas)
```
Comment (lpabon) [inline pkg/cluster/px/px.go]: May want to add a comment to as what this is doing and why it is needed

File: pkg/cluster/px/px.go [hunk 2]
```diff
@@ -243,3 +256,217 @@ func (ops *pxClusterOps) runDockerPuller(imageToPull string) error {
...
+			}
+
+			dCopy = dCopy.DeepCopy()
+			if dCopy.Annotations == nil {
+				return nil, false, nil // done as this is not an app we touched
+			}
+
+			val, present := dCopy.Annotations[replicaMemoryKey]
+			if !present || len(val) == 0 {
+				return nil, false, nil // done as this is not an app we touched
```
Comment (lpabon) [inline pkg/cluster/px/px.go]: May want to log if it is not present, just in case someone deleted it.

File: pkg/cluster/px/px.go [hunk 3]
```diff
@@ -243,3 +256,217 @@ func (ops *pxClusterOps) runDockerPuller(imageToPull string) error {
...
+	var sharedDeps []apps_api.Deployment
+	var sharedSS []apps_api.StatefulSet
+	for _, sc := range scs.Items {
+
+		params, err := ops.k8sOps.GetStorageClassParams(&sc)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		if isShared, present := params["shared"]; present && isShared == "true" {
```
Comment (lpabon) [inline pkg/cluster/px/px.go]: Cool

File: pkg/cluster/px/px.go (hunk 1 above)
Comment (harsh-px) [reply pkg/cluster/px/px.go]: done

File: pkg/cluster/px/px.go (hunk 2 above)
Comment (harsh-px) [reply pkg/cluster/px/px.go]: done

**Pull request merged**